lms/djangoapps/bookmarks/services.py

```diff
@@ -31,7 +59,13 @@ def bookmarks(self, course_key):
         Returns:
             list of dict:
         """
-        return api.get_bookmarks(self._user, course_key=course_key, fields=DEFAULT_FIELDS + OPTIONAL_FIELDS)
+        bookmarks = api.get_bookmarks(self._user, course_key=course_key, fields=DEFAULT_FIELDS + OPTIONAL_FIELDS)
+
+        if course_key.run is not None:
```

Comment:
Well, you might want to use `fill_in_run` and then check. And if the run can't be filled in, then that means that you don't have a real course, so I think I'd lean towards not returning any bookmarks.